components/ILIAS/Forum/classes/Tables/ForumStatisticsTable.php

```diff
@@ -0,0 +1,240 @@
+{
+    private UIFactory $ui_factory;
+    private ilLanguage $lng;
+    private bool $has_active_lp = false;
+    /** @var int[] */
+    private array $completed = [];
+    /** @var int[] */
+    private array $failed = [];
+    /** @var int[] */
+    private array $in_progress = [];
+    private HttpRequest $request;
+    private UIRenderer $ui_renderer;
+
+    public function __construct(
+        private readonly ilObjForum $forum,
+        private readonly ilForumProperties $obj_properties,
+        private readonly bool $has_general_lp_access,
+        private readonly bool $has_rbac_or_position_access,
+        private readonly ilObjUser $actor,
+    ) {
+        global $DIC;
+        $this->ui_factory = $DIC->ui()->factory();
+        $this->ui_renderer = $DIC->ui()->renderer();
+        $this->request = $DIC->http()->request();
+        $this->lng = $DIC->language();
+
+        $lp = ilObjectLP::getInstance($forum->getId());
+        if ($lp->isActive()) {
+            $this->has_active_lp = true;
+        }
+
+        if ($this->has_active_lp && $this->has_general_lp_access) {
+            $this->lng->loadLanguageModule('trac');
+            $this->completed = ilLPStatusWrapper::_lookupCompletedForObject($forum->getId());
+            $this->in_progress = ilLPStatusWrapper::_lookupInProgressForObject($forum->getId());
+            $this->failed = ilLPStatusWrapper::_lookupFailedForObject($forum->getId());
+        }
+    }
+
+    public function getComponent(): DataTable
+    {
+        return $this->ui_factory->table()
+                                ->data(
+                                    $this->lng->txt('frm_moderators'),
+                                    $this->getColumns(),
+                                    $this->getDataRetrieval()
+                                )
+                                ->withRequest($this->request);
+    }
+
+    protected function getColumns(): array
+    {
+        $columns = [
+            'ranking' => $this->ui_factory->table()->column()->text(
+                $this->lng->txt('frm_statistics_ranking')
+            )->withIsSortable(true),
+            'login' => $this->ui_factory->table()->column()->text(
+                $this->lng->txt('login')
+            )->withIsSortable(true),
+            'lastname' => $this->ui_factory->table()->column()->text(
+                $this->lng->txt('lastname')
+            )->withIsSortable(true),
+            'firstname' => $this->ui_factory->table()->column()->text(
+                $this->lng->txt('firstname')
+            )->withIsSortable(true),
+        ];
+        if ($this->has_active_lp && $this->has_general_lp_access) {
+            $columns['progress'] = $this->ui_factory->table()->column()->text(
+                $this->lng->txt('learning_progress')
+            )->withIsSortable(false);
+        }
+        return $columns;
+    }
+
+    protected function getDataRetrieval(): DataRetrieval
+    {
+        $data_retrieval = new class (
```

Comment:
I suggest to drop this "Anonymous Class Approach" and implement the `DataRetrieval` interface by the `ForumStatisticsTable` itself. The problem I have with this in this specific case is the number of arguments you have to pass through to the constructor of the anonyous class. My suggested approach simplifies this a lot (IMO).